pkg/operator/controller/status/controller.go

```diff
@@ -299,55 +348,63 @@ func computeOperatorProgressingCondition(dnsAvailable bool, oldVersions, curVers
 		switch opv.Name {
 		case OperatorVersionName:
 			if opv.Version != operatorReleaseVersion {
+				status = configv1.ConditionTrue
+				progressingReasons = append(progressingReasons, "UpgradingOperator")
 				messages = append(messages, fmt.Sprintf("Moving to release version %q.", operatorReleaseVersion))
-				progressing = true
 			}
 		case CoreDNSVersionName:
 			if opv.Version != coreDNSImage {
+				status = configv1.ConditionTrue
+				progressingReasons = append(progressingReasons, "UpgradingCoreDNS")
 				messages = append(messages, fmt.Sprintf("Moving to coredns image version %q.", coreDNSImage))
-				progressing = true
 			}
 		case OpenshiftCLIVersionName:
 			if opv.Version != openshiftCLIImage {
+				status = configv1.ConditionTrue
+				progressingReasons = append(progressingReasons, "UpgradingOpenShiftCLI")
 				messages = append(messages, fmt.Sprintf("Moving to openshift-cli image version %q.", openshiftCLIImage))
-				progressing = true
 			}
 		case KubeRBACProxyName:
 			if opv.Version != kubeRBACProxyImage {
+				status = configv1.ConditionTrue
+				progressingReasons = append(progressingReasons, "UpgradingKubeRBACProxy")
 				messages = append(messages, fmt.Sprintf("Moving to kube-rbac-proxy image version %q.", kubeRBACProxyImage))
-				progressing = true
 			}
 		}
 	}
 
-	if progressing {
-		progressingCondition.Status = configv1.ConditionTrue
-		progressingCondition.Reason = "Reconciling"
+	if len(progressingReasons) != 0 {
+		progressingCondition.Status = status
+		progressingCondition.Reason = strings.Join(progressingReasons, "And")
+		progressingCondition.Message = strings.Join(messages, "\n")
 	} else {
 		progressingCondition.Status = configv1.ConditionFalse
 		progressingCondition.Reason = "AsExpected"
-	}
-	progressingCondition.Message = dnsEqualConditionMessage
-	if len(messages) > 0 {
-		progressingCondition.Message = strings.Join(messages, "\n")
+		progressingCondition.Message = dnsEqualConditionMessage
 	}
 
 	return progressingCondition
 }
 
 // computeOperatorAvailableCondition computes the operator's current Available status state.
-func computeOperatorAvailableCondition(dnsAvailable bool) configv1.ClusterOperatorStatusCondition {
+func computeOperatorAvailableCondition(haveDNS bool, dns *operatorv1.DNS) configv1.ClusterOperatorStatusCondition {
 	availableCondition := configv1.ClusterOperatorStatusCondition{
 		Type: configv1.OperatorAvailable,
 	}
-	if dnsAvailable {
-		availableCondition.Status = configv1.ConditionTrue
-		availableCondition.Reason = "AsExpected"
-		availableCondition.Message = "DNS default is available"
-	} else {
+
+	switch {
+	case !haveDNS:
+		availableCondition.Status = configv1.ConditionFalse
+		availableCondition.Reason = "DNSDoesNotExist"
+		availableCondition.Message = `DNS "default" does not exist.`
```

Comment:
Should we consider using `dns.Name` in place of `"default"`?

Reply:
In this case (`case !haveDNS`), `dns` is nil.  I could use `dns.Name` in the other cases, but the operator doesn't reconcile any dns other than "default" anyway, so it seemed just as well to use a string literal throughout.

Reply:
Fair enough.